src/libs/tools/benchmarks/benchmark_crypto_comparison.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file
+ *
+ * @brief benchmark for getenv
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ *
+ */
+
+#include <backendbuilder.hpp>
+#include <kdbconfig.h>
+#include <kdbtimer.hpp>
+
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+
+extern "C" char ** environ;
+
+// test key is located at src/plugins/crypto/test_key.asc
+#define GPG_TEST_KEY_ID "DDEBEF9EE2DC931701338212DAF635B17F230E8D"
+
+enum PluginVariant
+{
+	NO_CRYPTO = 0,
+	CRYPTO_OPENSSL = 1,
+	CRYPTO_GCRYPT = 2,
+	CRYPTO_BOTAN = 3
+};
+
+static const std::string plugin_variant_names[] = { "no_crypto_plugin", "crypto_openssl", "crypto_gcrypt", "crypto_botan" };
+
+long long nr_keys = 100LL; // should be higher but crypto performance is actually quite bad atm
+
+const int benchmarkIterations = 11; // is a good number to not need mean values for median
+
+kdb::KeySet toWrite;
+
+template <int VARIANT>
```

Comment:
It is identical (at least in C), but PluginVariant would be clearer here instead of int.